### PR TITLE
feat: add fallback to GitHub API repo search and use correct auth

### LINKED ISSUE #1440: Bug: AI generation of highlight form body throws 401 on getting commits from github
### Describe the bug

When attempting to create an Ai generated text body for my highlight, the icon spins indefinitely and never generates the text:

Screenshot 2023-07-26 at 6 02 35 PM

I do see in the console that a `GET` https://api.github.com/repos/open-sauced/api/pulls/223/commits (for this specific highlight) throws a 401, not authorized:

```json
{
	"message": "Bad credentials",
	"documentation_url": "https://docs.github.com/rest"
}
```

Very weird that this would send a 401 since I can curl this on my own machine just fine without having to provide any auth:
```sh
❯ curl https://api.github.com/repos/open-sauced/api/pulls/223/commits
[
  {
    "sha": "9a0b8bd59c5ffb1d60ef5f1277329b9217d00af3",
     ...
   }
]
```

### Steps to reproduce

1. Go to highlights
2. Attempt to input a PR and generate text
3. Open dev console and see response to `api.github.com` for that pull request

### Browsers

Firefox

### Additional context (Is this in dev or production?)

_No response_

### Code of Conduct

- [X] I agree to follow this project's Code of Conduct

### Contributing Docs

- [X] I agree to follow this project's Contribution Docs

### PR DESCRIPTION
**Pull request by brandonroberts**
## Description

- Adds a fallback when using the GitHub API endpoint for repositories. If the rate limit is reach, it uses our `/repos/search` endpoint to return the data
- Also fixes some GitHub token issues with not providing the correct bearer token. This impacted the highlight form for AI generation when getting commits from GitHub
- Now the token is added to the headers only if present

## What type of PR is this? (check all applicable)

- [x] 🍕 Feature
- [ ] 🐛 Bug Fix
- [ ] 📝 Documentation Update
- [ ] 🎨 Style
- [ ] 🧑‍💻 Code Refactor
- [ ] 🔥 Performance Improvements
- [ ] ✅ Test
- [ ] 🤖 Build
- [ ] 🔁 CI
- [ ] 📦 Chore (Release)
- [ ] ⏩ Revert

## Related Tickets & Documents

Closes #1440 

## Steps to QA

1. Log in
2. Navigate to `/feed`
3. Open the highlight input form
4. Note the highlight suggestions returned below

## Added to documentation?

- [ ] 📜 README.md
- [ ] 📓 docs.opensauced.pizza
- [ ] 🍕 dev.to/opensauced
- [ ] 📕 storybook
- [ ] 🙅 no documentation needed

### CODE REVIEW

**Review by nickytonline — APPROVED** [review]
🚢

File: lib/hooks/useSearchRepos.ts [hunk 1]
```diff
@@ -1,7 +1,68 @@
 import useSWR, { Fetcher } from "swr";
-import githubApiFetcher from "lib/utils/github-api-fetcher";
+import { supabase } from "lib/utils/supabase";
+
+const baseUrl = "https://api.github.com";
```
Comment (jpmcb) [inline lib/hooks/useSearchRepos.ts:4]: Should this use the env found in `NEXT_PUBLIC_API_URL`?

**Review by jpmcb — APPROVED** [review]

File: lib/hooks/useSearchRepos.ts (hunk 1 above)
Comment (brandonroberts) [reply lib/hooks/useSearchRepos.ts:4]: Not on this one because this is the base for the GitHub API and not ours

**Review by bdougie — APPROVED** [review]

**Pull request merged**